matrix_sdk_base/src/models/room.rs

```diff
@@ -1764,4 +1767,54 @@ mod test {
         assert_eq!(encryption_info.rotation_period(), 100_000);
         assert_eq!(encryption_info.rotation_period_messages(), 100);
     }
+
+    #[test]
+    fn power_level_overflow() {
+        let room_id = get_room_id();
+        let user_id = user_id!("@example:localhost");
+
+        let content = MemberEventContent {
+            avatar_url: None,
+            displayname: Some("user1".into()),
+            is_direct: Some(false),
+            membership: MembershipState::Join,
+            third_party_invite: None,
+        };
+        let member = SyncStateEvent {
+            event_id: event_id!("$h29iv0s8:example.com"),
+            origin_server_ts: SystemTime::now(),
+            sender: user_id.clone(),
+            state_key: "@example:localhost".into(),
+            unsigned: Unsigned::default(),
+            content,
+            prev_content: None,
+        };
+
+        let mut room_member = RoomMember::new(&member, &room_id);
+
+        // This power level was found in the DayDream client to overflow with the
+        // previous normalization logic
+        let mut content = PowerLevelsEventContent::default();
+        *content
+            .users
+            .entry(user_id.clone())
+            .or_insert_with(|| js_int::Int::new(4503599627370495).unwrap()) =
```

Comment:
Can't we use the `int!()` macro here?

Reply:
The `int!` macro takes i32 so the number is too big unfortunatly.

Reply:
Ah, ok.